### PR TITLE
Update daal version and naming

### PR DESCRIPTION
**Pull request by Pahandrovich**
Checklist
* [X] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [X] Reset the build number to `0` (if the version changed)
* [X] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [X] Ensured the license file is being packaged.

### CODE REVIEW

File: recipe/meta.yaml [hunk 1]
```diff
@@ -304,6 +304,86 @@ outputs:
       commands:
         - ls -A $PREFIX/lib/*  # [unix]
         - ls -A $PREFIX/include/*  # [unix]
+  
+  - name: daal
+    script: repack.sh   # [unix]
```
Comment (napetrov) [inline recipe/meta.yaml:309]: I don't think we need to call repack for this empty metapackage

**Review by beckermr — CHANGES_REQUESTED** [review]
Some of the run exports here look a little funny. The `daal` doesn't need a run export on `dal` since the original `daal` package didn't have any run exports.

**Review by beckermr — CHANGES_REQUESTED** [review]
It looks like you cleared the run requirements when you should have cleared the run_export sections.

File: recipe/meta.yaml [hunk 2]
```diff
@@ -1,8 +1,8 @@
-{% set mkl_version = "2020.4" %}
-{% set version = "2020.3" %}
-{% set buildnum = "304" %}  # [linux]
-{% set buildnum = "301" %}  # [osx]
-{% set buildnum = "311" %}  # [win]
+{% set mkl_version = "2021.1" %}
+{% set version = "2021.1" %}
+{% set buildnum = "0" %}  # [linux]
```
Comment (napetrov) [inline recipe/meta.yaml:3]: we have buildnums avaialbe now. 
But we anyway need need new TBB in conda-forge first.

**Comment by PetrovKP** [comment]
I think REAMDE needs to be updated

**Comment by napetrov** [comment]
@conda-forge-admin, please rerender

File: recipe/meta.yaml [hunk 3]
```diff
@@ -1,40 +1,55 @@
...
 package:
   name: intel_repack
   version: {{ version }}
 
 source:
-  - url: https://anaconda.org/intel/mkl/{{ mkl_version }}/download/{{ target_platform }}/mkl-{{ mkl_version }}-intel_{{ buildnum }}.tar.bz2
+  - url: https://anaconda.org/intel/mkl/{{ mkl_version }}/download/{{ target_platform }}/mkl-{{ mkl_version }}-intel_{{ mkl_buildnum }}.tar.bz2
     folder: mkl
-  - url: https://anaconda.org/intel/mkl-devel/{{ mkl_version }}/download/{{ target_platform }}/mkl-devel-{{ mkl_version }}-intel_{{ buildnum }}.tar.bz2
+    sha256:
```
Comment (napetrov) [inline recipe/meta.yaml:29]: Let's remove sha256 field. It can't be empty

File: recipe/meta.yaml [hunk 4]
```diff
@@ -278,20 +293,96 @@ outputs:
...
+  - name: daal-devel
+    build:
+      number: {{ buildnum|int + dstbuildnum|int }}
     requirements:
       run:
+        - {{ pin_subpackage('dal-devel', exact=True) }}
         - {{ pin_subpackage('daal-include', exact=True) }}
         - {{ pin_subpackage('daal', exact=True) }}
     about:
       home: https://software.intel.com/en-us/daal
```
Comment (napetrov) [inline recipe/meta.yaml:384]: Please pickup new names links - as example https://github.com/conda-forge/intel_repack-feedstock/pull/14

**Comment by isuruf** [comment]
Ping on this @Pahandrovich

**Comment by napetrov** [comment]
@conda-forge-admin, please rerender

**Review by napetrov — APPROVED** [review]

**Comment by napetrov** [comment]
@isuruf  - thanks for update and fixes. 
I think this is ready to go now

**Pull request merged**